pex/pip/tool.py

```diff
@@ -409,6 +414,10 @@ def _spawn_pip_isolated(
 
         command = pip_args + list(args)
 
+        # Append any pass-through pip args.
```

Comment:
The motivating option is a global option; so you can't pass it here. It would have to be above before `args` is appended (see: https://pip.pypa.io/en/stable/cli/pip/#cmdoption-keyring-provider).

For this sort of reason, which neither using Pip config files nor env vars has a similar ordering issue, I'm leery of a generic Pip flag pass through mechanism. I'd be happy to support `--keyring-provider` 1st class though if that works for you.

Reply:
Ok, looks like global options, including `--keyring-provider` can be passed in any position; so I have this wrong. I'd still be more in favor of a dedicated `--keyring-provider` pass through though since it seems likely a user would pass an option that, say `pip download` or `pip install` takes but `pip debug` does not. Since Pex does, in fact call `pip debug` to implement `--platform`, this would cause a surprising failure.

Reply:
> I'd be happy to support `--keyring-provider` 1st class though if that works for you.

The overall motivating use case is a change in Pants to pass `--keyring-provider` through. Will rework as a `--keyring-provider` option as suggested.

> Since Pex does, in fact call pip debug to implement --platform, this would cause a surprising failure.

Good to know.

Reply:
Also, this is still a WIP since I am having trouble running tests locally and am going to rely on CI to run the tests for me.

Reply:
You can try substituting `tox` with `./dtox.sh` in any tox command line fwiw.